OpenKeychain/src/main/java/org/sufficientlysecure/keychain/service/BackupNotificationService.java

```diff
@@ -0,0 +1,88 @@
+package org.sufficientlysecure.keychain.service;
+
+import android.app.AlarmManager;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+
+import org.sufficientlysecure.keychain.receiver.BackupNotificationAlarmReceiver;
+import org.sufficientlysecure.keychain.util.Preferences;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * This class is used to setup a backup reminder for every new key created by user
+ */
+public class BackupNotificationService extends Service {
```

Comment:
Yup, you are right. Initially, I thought it will be better to not run this on the UI thread (using AsyncTask or service) and created a service.  Instead, I should simply rename this class to BackupNotificationUtil and put in util.